sycl/include/CL/sycl/usm/usm_allocator.hpp

```diff
@@ -107,35 +122,26 @@ class usm_allocator {
         "Device pointers do not support address on host");
   }
 
-  // Allocate memory
-  template <
-      usm::alloc AllocT = AllocKind,
-      typename std::enable_if<AllocT == usm::alloc::host, int>::type = 0>
+  /// Allocates memory.
+  ///
+  /// @param Size is a count of elements to allocate memory for.
   pointer allocate(size_t Size) {
-    auto Result = reinterpret_cast<pointer>(detail::usm::alignedAllocHost(
-        getAlignment(), Size * sizeof(value_type), mContext, AllocKind));
-    if (!Result) {
-      throw memory_allocation_error();
-    }
-    return Result;
-  }
 
-  template <usm::alloc AllocT = AllocKind,
-            typename std::enable_if<AllocT != usm::alloc::host, int>::type = 0>
-  pointer allocate(size_t Size) {
     auto Result = reinterpret_cast<pointer>(
-        detail::usm::alignedAlloc(getAlignment(), Size * sizeof(value_type),
-                                  mContext, mDevice, AllocKind));
+        aligned_alloc(getAlignment(), Size * sizeof(value_type),
+                                 MDevice, MContext, AllocKind));
     if (!Result) {
       throw memory_allocation_error();
     }
     return Result;
   }
 
-  // Deallocate memory
-  void deallocate(pointer Ptr, size_t size) {
+  /// Deallocates memory.
+  ///
+  /// @param Ptr is a pointer to memory being deallocated.
+  void deallocate(pointer Ptr, size_t Size) {
```

Comment:
`Size` is not documented.
What do we need `Size` for?
Is it possible to deallocate a part of the previously allocated memory?

Reply:
https://en.cppreference.com/w/cpp/memory/allocator/deallocate

Symmetry?

Reply:
@bader cppreference says:
> **n** - number of objects earlier passed to allocate

I'm not sure if this method is designed to do partial deallocations. @jbrodman can you please clarify?